play/src/front/Phaser/Game/MapEditor/Tools/ExplorerTool.ts

```diff
@@ -0,0 +1,311 @@
+        // Define new zone to zoom
+        if (this.downIsPressed || this.upIsPressed || this.leftIsPressed || this.rightIsPressed)
+            this.defineZoomToCenterCameraPosition();
+        if (event.key === "ArrowDown" || event.key === "s") {
+            this.downIsPressed = false;
+        }
+        if (event.key === "ArrowUp" || event.key === "w" || event.key === "z") {
+            this.upIsPressed = false;
+        }
+        if (event.key === "ArrowLeft" || event.key === "q" || event.key === "a") {
+            this.leftIsPressed = false;
+        }
+        if (event.key === "ArrowRight" || event.key === "d") {
+            this.rightIsPressed = false;
+        }
+        this.mapEditorModeManager.handleKeyDownEvent(event);
+    };
+    private wheelHandler = (
+        pointer: Phaser.Input.Pointer,
+        gameObjects: Phaser.GameObjects.GameObject[],
+        deltaX: number,
+        deltaY: number,
+        deltaZ: number
+    ) => {
+        // Calculate the velocity of the scroll
+        const velocity = deltaY / 53;
+        const zoomFactor = 1 - velocity * 0.1;
+        // Restore camera mode
+        this.scene.zoomByFactor(zoomFactor, velocity);
+    };
+    private pointerDownHandler = (pointer: Phaser.Input.Pointer) => {
+        this.explorationMouseIsActive = true;
+        this.scene.input.setDefaultCursor("grabbing");
+    };
+    private pointerMoveHandler = (pointer: Phaser.Input.Pointer) => {
+        if (!this.explorationMouseIsActive) return;
+        this.scene.cameras.main.scrollX -= pointer.velocity.x / 10;
+        this.scene.cameras.main.scrollY -= pointer.velocity.y / 10;
+
+        // Define new zone to zoom
+        this.defineZoomToCenterCameraPosition();
+
+        this.scene.markDirty();
+    };
+    private pointerUpHandler = (pointer: Phaser.Input.Pointer, gameObjects: Phaser.GameObjects.GameObject[]) => {
+        this.scene.input.setDefaultCursor("grab");
+        this.explorationMouseIsActive = false;
+
+        if (gameObjects.length > 0) {
+            const gameObject = gameObjects[0];
+            if (gameObject instanceof Entity || gameObject instanceof AreaPreview)
+                mapExplorationObjectSelectedStore.set(gameObject);
+        }
+
+        this.scene.markDirty();
+    };
+
+    constructor(private mapEditorModeManager: MapEditorModeManager) {
+        this.scene = gameManager.getCurrentGameScene();
+        this.entitiesManager = this.scene.getGameMapFrontWrapper().getEntitiesManager();
+    }
+
+    public update(time: number, dt: number): void {
+        if (this.downIsPressed) {
+            this.scene.cameras.main.scrollY += 10;
+        }
+        if (this.upIsPressed) {
+            this.scene.cameras.main.scrollY -= 10;
+        }
+        if (this.leftIsPressed) {
+            this.scene.cameras.main.scrollX -= 10;
+        }
+        if (this.rightIsPressed) {
+            this.scene.cameras.main.scrollX += 10;
+        }
```

Comment:
Instead of moving the camera from 10 pixel games, we should move the camera from an amount proportional to the zoom level.

If we move from 10 pixel games, the movement is too fast when we are zoomed in and too slow when we are zoomed out.
We should move from a constant amount of "screen pixels", not "game pixels".